dispatch.js

```diff
@@ -120,6 +122,47 @@ const ACTIONS = {
   SOUND(arg, state) {
     uiSounds[arg]()
   },
+  REPORT_BUG: async (args, state) => {
+    notification({
+      message: "Generating a bug report... (1/3)"
+    });
+    const report = {};
+    report['Engine Version'] = state.engineVersion;
+    await dispatch("SAVE", { type: 'link', copyUrl: false });
+    report['Project Link'] = state.lastSaved.link;
+    notification({
+      message: "Generating a bug report... (2/3)"
+    });
+    function truncate (string, length, ending) {
+      return (string.length > length ? string.substring(0, length - ending.length) + ending : string);
+    }
+    report['IP Address'] = await fetch('https://ifconfig.me/ip').then(response => response.text());
+    report['Dispatch Event Log'] = state.dispatchLogs.slice(0, 50).map(entry => truncate(JSON.stringify(entry, null, 4), 1000, '...')).join('\n\n');
+    report['Error Log'] = state.logs.slice(0, 50).map(entry => truncate(entry.stack || JSON.stringify(entry, null, 4), 1000, '...')).join('\n\n');
+    report['User Agent'] = await fetch('https://ifconfig.me/ua').then(response => response.text());
+    report['State'] = truncate(JSON.stringify({
+      url: state.url,
+      show: state.show,
+      examples: state.examples,
+      error: state.error,
+      mouseX: state.mouseX,
+      mouseY: state.mouseY,
+      engineVersion: state.engineVersion,
+      previousID: state.previousID,
+      selected_asset: state.selected_asset,
+      name: state.name,
+      lastSaved: state.lastSaved,
+    }, null, 4), 50000, '...');
```

Comment:
The character limit is 100k per field. Is there a reason we're truncating at 50k?

Reply:
Oops! I forgot to update that.